cns/ipampool/monitor.go

```diff
@@ -101,6 +107,11 @@ func (pm *Monitor) Start(ctx context.Context) error {
 			subnetCIDR = nnc.Status.NetworkContainers[0].SubnetAddressSpace
 			subnetARMID = GenerateARMID(&nnc.Status.NetworkContainers[0])
 
+			// Add Primary IP to Map, if not present.
+			for i := 0; i < len(nnc.Status.NetworkContainers); i++ {
+				ipAddresses[nnc.Status.NetworkContainers[i].PrimaryIP] = 1
+			}
```

Comment:
I don't understand why we're setting a value of `1` here, can you explain?

Reply:
It is a hashmap, we need to set the value to something. The other alternative was to set it to true.

Reply:
Are you trying to use the `map` as a `Set`? The preferred way to do that is to make the Value type `interface{}` and set `ipAddresses[nnc.Status.NetworkContainers[i].PrimaryIP] = nil`, as that has no memory overhead